app/server.R

```diff
@@ -788,14 +789,28 @@ server <- function(input, output, session) {
     # Only show forecasters that have data for the score chosen
     updateForecasterChoices(session, df, input$forecasters, input$scoreType)
 
-    # If we are switching between coverage and other score types we need to
-    # update the as of data we have so it matches the correct locations shown
-    if (input$location == "US") {
+    showScoreExplanation(session, input$scoreType, DASH_SUFFIX)
+    USE_CURR_TRUTH <<- TRUE
+
+    # Normally when score type selection changes, we reuse the existing truth
+    # plot (behavior is toggled by `USE_CURR_TRUTH`). This is valid as long
+    # as the locations reported by each score type are the same.
+    #
+    # However, regardless of which location is selected, coverage aggregates
+    # scores over states that have forecasts reported by all selected
+    # forecasters. This means that when changing to or from "coverage" and
+    # any single state or "US" is selected as the location (that is, we are
+    # not aggregating over shared locations), we need to regenerate the truth
+    # plot with updated truth data.
+    if ((input$scoreType == "coverage" || PREV_SCORE_TYPE == "coverage") && input$location != TOTAL_LOCATIONS) {
       updateAsOfData()
+      # Don't ONLY create a new truth plot, make a new main plot, too.
+      RE_RENDER_TRUTH <<- FALSE
```

Comment:
praise: explanatory comment is a good call

(i'm sure this variable name makes sense in context, just noting the 🙃 that setting RE_RENDER to FALSE actually causes the plot to re-render)

Reply:
Yes, I was confused by the names too, they deserve a refactor.